### DISCUSSION TITLE
EdgeX compatibility with RPC broker

### DISCUSSION THREAD

**Discussion by dipshirajput** [edgexfoundry/community#147] [Q&A]
Hi All,

1. can we configure EdgeX with RPC broker ? if yes, which RPC broker it can use ? 
2. Can we use IoTDB with EdgeX ?

please let me know for any suggestions. Thanks in advance.

**Comment by lenny-goodell** [comment]
> Hi All,
> 
> 1. can we configure EdgeX with RPC broker ? if yes, which RPC broker it can use ?

Nope, EdgeX currently only supports Redis Pub/Sub (default) , MQTT or NATS (must select at build time) for the MessageBus
Please see https://docs.edgexfoundry.org/3.0/microservices/general/messagebus for more details

> 2. Can we use IoTDB with EdgeX ?
> 

Not out of the box. This is an interface for the DB which you could implement for IoTDB. You would have to do this in you own fork EdgeX repos

**Comment by andresrinivasan** [comment]
If you're trying to get the Event data into IoTDB, you might consider writing an application service that publishes the data.

**Reply by lenny-goodell** [reply]
Yes a custom App Service could be written which consume the Events from the EdgeX MessageBus and stores them into in IoTDB.
